File: JavaScript/Pessoa_POO.js
// 4. Orientação a Objetos:

// JavaScript (Prototipagem e Flexibilidade): JavaScript permite criar objetos sem classes, usando protótipos ou a palavra-chave class.

class Pessoa_POO {
    constructor(nome) {
        this.nome = nome;
    } saudacao() {
        console.log("Olá,meu nome é " + this.nome);
    }
}
const pessoa = new Pessoa_POO("Carlos");
pessoa.saudacao();

File: Java/Pessoa_POO.java
// 4. Orientação a Objetos:

// Java (Classes e Objetos): Java usa classes e objetos para orientação a objetos de forma mais rígida.

public class Pessoa_POO {
    String nome;

    public Pessoa_POO(String nome) {
        this.nome = nome;
    }

    public void saudacao() {
        System.out.println("Olá, meu nome é " + nome);
    }

    public static void main(String[] args) {
        Pessoa_POO pessoa = new Pessoa_POO("Carlos");
        pessoa.saudacao();
    }
}
